### PR TITLE
Added:DiscountCode APIs attributes, relations and permissions

### LINKED ISSUE #3735: Implement Discount Codes API with attributes, relationships and permissions
To be done after #3728, #3720.

### PR DESCRIPTION
**Pull request by enigmaeth**
Follow-up to PR #3769.

Fixes #3735.

### CODE REVIEW

**Comment by enigmaeth** [comment]
@shubham-padia 
Marketer_id and user relationships have been removed. 
(The original PR got closed accidentally. Added in the description : PR #3769)

File: app/api/events.py [hunk 1]
```diff
@@ -189,7 +198,19 @@ def before_get_object(self, view_kwargs):
                 else:
                     view_kwargs['id'] = None
 
-    decorators = (is_coorganizer, )
```
Comment (shubham-padia) [inline app/api/events.py]: Why was this removed

File: app/api/events.py [hunk 2]
```diff
@@ -189,7 +198,19 @@ def before_get_object(self, view_kwargs):
...
+        if view_kwargs.get('discount_code_id') is not None:
+            try:
+                discount_code = self.session.query(DiscountCode).filter_by(id=view_kwargs['discount_code_id']).one()
+            except NoResultFound:
+                raise ObjectNotFound({'parameter': 'discount_code_id'},
+                                     "DiscountCode: {} not found".format(view_kwargs['discount_code_id']))
+            else:
+                if discount_code.marketer_id is not None:
+                    view_kwargs['id'] = discount_code.marketer_id
```
Comment (shubham-padia) [inline app/api/events.py]: @enigmaeth Can you recheck this. I see marketer_id here

Comment (enigmaeth) [inline app/api/events.py]: Resolved.
It should be event_id here.
